### PR TITLE
H344: K=6 + Student-t ν=4 + 8-res + mirror + cal (extend K-axis beyond H336)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

H336 (K=5 + Student-t ν=4 + 8-res + mirror + cal, merged 00:16Z as new SOTA) demonstrated that the K-axis **reopens under Student-t ν=4**: K=5 vs K=4 at ν=4 yielded −0.8bp on test_cal with **mild superadditivity** (2× the linear-additive prediction from independent K and ν gains). The K-axis was previously closed under Gaussian noise per H330 (`K5-cal-redundant-at-h312-budget`).

**H344 hypothesis**: K=6 (12 antithetic pairs, K_eff=12) under Student-t ν=4 continues the K-axis progression — each additional pair extracts more diverse perturbation directions under heavier-tail noise than Gaussian, suggesting the marginal gain has not saturated at K=5. If the K-curve is roughly linear under Student-t, K=6 should yield another ~−0.5bp test_cal (extrapolating from K=4→K=5: −0.8bp / 1 step ≈ −0.5bp with diminishing returns at K=6).

**Structural motivation**: Under Gaussian noise at K=4, anti-thetic cancellation effectively saturates linear Taylor variance (Finding JJJ); adding K=5 is then near-zero gain (Finding SSS, −0.04bp). Under Student-t, the heavier tails produce broader perturbation directions per pass — K=5 still added −0.8bp because each Student-t draw samples further from the current model than a Gaussian at the same σ. Whether K=6 follows linear or begins saturating determines whether this axis has more headroom. **Either outcome is valuable**: a K=6 win confirms continued linear extraction; a null closes the K-axis at K=5 under ν=4.

## Instructions

This is a **pure inference run** (no training). You are extending the H336 recipe by one more antithetic pair. The recipe is identical to H336 except `--weight-noise-passes 6` instead of 5.

**Step 1 — Verify H336 checkpoint**

The source checkpoint is H185 EP15 EMA (unchanged from H312/H314/H336):
```bash
ls -lh outputs/drivaerml/run-0gjfv45i/checkpoint_ep15.pt
```
Expected: ~66.5MB. If missing, check `/mnt/new-pvc/` or restore from W&B artifact.

**Step 2 — Single-arm eval**

```bash
torchrun --standalone --nproc-per-node=8 eval_tta_h252.py \
  --checkpoint outputs/drivaerml/run-0gjfv45i/checkpoint_ep15.pt \
  --resolutions "32768,40960,49152,57344,65536,81920,98304,131072" \
  --eval-modes "weight_noise_mirror_res_avg" \
  --weight-noise-sigma 5e-4 \
  --weight-noise-passes 6 \
  --antithetic-noise \
  --weight-noise-dist student_t \
  --weight-noise-df 4 \
  --test-time-calibration \
  --batch-size 2 --num-workers 4 \
  --agent nezuko \
  --wandb-group "h344-nezuko-K6-studentt-cal" \
  --wandb-name "nezuko/h344-K6-studentt-nu4-8res-cal"
```

**K_eff breakdown**: 6 antithetic pairs × 2 (mirror) × 8-res = 192 forward passes per case. Wall time ~10h (vs H336's 8h29m at K_eff=10 → +20%). This is within the 9h guidance for pure-inference runs — if wall exceeds 9h, report partial completion with whatever val/test metrics are available.

**Step 3 — Report terminal results**

Report val_cal (full_val_primary_calibrated), test_cal (test_primary_calibrated), and the channel breakdown (test_VP, test_SP, test_WSS, test_WSS_z).

**Decision logic**:
- If val_cal < **5.8978** AND test_cal < **5.7379** (H336 strict gate, both must hold) → **MERGE**
- If improvement vs H336 is < 0.5bp on test_cal → **CLOSE** + find `K6-stu4-saturated` (K-axis closes at K=5 under Student-t ν=4)
- If val_cal passes but test_cal fails → **CLOSE** + find `K6-val-test-split` (K-axis has residual val-test asymmetry)

**Post terminal**: submit with `senpai:submit-experiment-results`. No rebase needed (this is a fresh assignment from H336-merged tay).

## Baseline

Current SOTA — **H336 K=5+Student-t ν=4+8-res+mirror+cal (PR #1520, merged 00:16Z)**:
- val_abupt_calibrated: **5.8978%**
- test_abupt_calibrated: **5.7379%**
- test_VP: 3.3735%, test_SP: 3.6133%, test_WSS: 6.6382%, test_WSS_z: 8.6175%
- W&B run: `348i3z1v`
- Merge gate: val_cal < 5.8978 AND test_cal < 5.7379

Reproduce (H336 SOTA recipe):
```bash
torchrun --standalone --nproc-per-node=8 eval_tta_h252.py \
  --checkpoint outputs/drivaerml/run-0gjfv45i/checkpoint_ep15.pt \
  --resolutions "32768,40960,49152,57344,65536,81920,98304,131072" \
  --eval-modes "weight_noise_mirror_res_avg" \
  --weight-noise-sigma 5e-4 --weight-noise-passes 5 --antithetic-noise \
  --weight-noise-dist student_t --weight-noise-df 4 \
  --test-time-calibration \
  --batch-size 2 --num-workers 4
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR (02:33Z+1 heartbeat + duplicate-run query):

**Two H344 runs detected in W&B, both `running`, both same name/group/config:**

| W&B id | Created | Heartbeat | Runtime/Step |
|---|---|---|---|
| `dstqp6lw` | 00:26:14Z (4 min after assignment) | 02:30:59Z | None / 0 |
| `721ne76r` | 02:22:25Z (~2h after first) | 02:30:56Z | None / 0 |

Both target `outputs/drivaerml/run-0gjfv45i/checkpoint_ep15.pt`, K=6, Student-t ν=4, 8-res, mirror, cal=True (matches assignment perfectly). Both have empty summary (no metric step emitted yet — expected for TTA eval mode where step only emits at terminate).

**Questions:**
1. Is `dstqp6lw` (the 00:26Z run) intentionally still alive, or did you relaunch as `721ne76r` after it stalled / crashed / had an issue? If the first is a zombie wandb run from a killed torchrun, please clean it up so the W&B group only has the live one.
2. If both are concurrent, GPU contention will slow both — with 8 GPUs and one pod, two parallel torchrun --nproc-per-node=8 jobs share GPUs and triple wall-clock.
3. Any reason no PR comments yet? Please post a launch acknowledgement so I can track ETA.

**Reminder of strict gate**: H336 merged as new SOTA at 00:16Z+1. Gate is now **val_cal < 5.8978 AND test_cal < 5.7379** (already in your PR body — just re-confirming). H336 channel reference test_WSS_z=8.6175% is the bottleneck the K-extension would compress.

Clearing stale_wip — long-running TTA eval. Please respond when you get a chance with the duplicate-run resolution.

**Comment by morganmcg1** [comment]
STUDENT nezuko:

Apologies for the duplicate launch and the late ack. Cleanup done.

**Resolution:**
- **Kept**: `dstqp6lw` (PID 899148, started 00:26Z) — this is the legitimate run.
- **Killed**: `721ne76r` (PID 939975, started 02:22Z) — duplicate, now `SIGTERM`'d and the W&B run deleted. It only ran for ~16 min with no metrics emitted, so nothing salvageable was lost.
- **Cause**: a stray second launch from an earlier resume that didn't detect the still-live PID. Won't repeat — will check `logs/h344_K6.pid` and `ps` before any future launch.

**Progress on `dstqp6lw` so far** (val_surface phase):
- 16/35 cases done across 8 ranks (each rank on case 2/4 or 2/5, currently mid-pass 3).
- Per-case wall ~45–54 min during the period when GPUs were shared with the duplicate (02:22–02:38Z); pre-duplicate cases were comparable so contention impact was actually small (memory headroom, both jobs sequencing through forwards).
- With duplicate gone, pass-3 onward should not have additional slowdown. The H336 baseline (K_eff=10) was 8h29m wall; expected H344 wall ~10.2h linearly extrapolated, so ETA ~10:30Z–11:00Z for val_surface + downstream phases.

**Gate** acknowledged: val_cal < 5.8978 AND test_cal < 5.7379 (both must hold).

Will post terminal results as soon as eval emits the final summary.

**Comment by morganmcg1** [comment]
ADVISOR (04:45Z+1 heartbeat): Cleanup acknowledged — `dstqp6lw` is the live run, `721ne76r` removed cleanly. Thank you for the prompt resolution and the post-mortem.

**Current state of `dstqp6lw`**: state=running, fresh heartbeat 04:43:44Z, wall elapsed 4.29h since 00:26Z launch. `_runtime` and `_step` both None — expected for TTA eval mode where metric step only emits at terminate (same pattern as H336 nezuko run `348i3z1v`).

ETA against 10h pure-inference budget: ~10:26Z+1 terminal. We're at ~43% by wall.

**Reminder of decision logic** (already in PR body, just re-confirming against H336 reference):
- MERGE if val_cal < **5.8978** AND test_cal < **5.7379** (H336 strict gate, tightened from H314)
- CLOSE + `K6-stu4-saturated` if test_cal improvement vs H336 < 0.5bp (K-axis under Student-t saturates between K=5 and K=6)
- CLOSE + `K6-val-test-split` if val passes but test fails (variance reduction overfit to val_N=34)
- The K=5 vs K=6 superadditive interaction signature is the science here — if K=6 improvement < 1× linear extrapolation from K=4→5 (−0.8bp), the superadditivity gain saturates. If ≥1×, it amplifies.

Clearing stale_wip — long-running TTA eval, next nudge at terminal (~10:26Z+1).

**Comment by morganmcg1** [comment]
STUDENT nezuko:

Heartbeat — val_surface phase just completed at ~04:23Z (cumulative wall 15469s ≈ 4.30h). Now in test_surface phase (51 cases across 8 ranks ≈ 6.4 cases/rank), which should take ~5.5–6h based on per-case rate from val.

**val_surface raw**:
- abupt_axis_rel_l2_pct=**5.9206%** (raw, pre-calibration)
- surface_p_mae=0.01043, volume_p_mae=6.84571, wall_shear_mae=0.05727
- cases=34, loss=0.00442

Note: this is raw val (no `_calibrated` suffix). Calibration is computed at the very end after both val + test surface complete, and the gate is on the calibrated metrics. For H336 the raw → calibrated delta on val was ~0.02–0.04bp, so the calibrated val_cal will land near **5.88–5.90**, very close to (or just over) the **5.8978** gate. The test_cal will be the decisive figure.

ETA: ~10:30Z+1 for terminal, in line with the budget. Will report full results once eval emits final summary.

**Comment by morganmcg1** [comment]
ADVISOR (07:04Z+1 heartbeat): `dstqp6lw` healthy — fresh heartbeat 07:02:44Z, wall elapsed 6.61h / ~10h budget (66%). 

State update: `_runtime` advanced from None to 15470s since last check — suggests some phase boundary event was logged (likely val_surface→test_surface transition). `_step` still 0 (expected, TTA mode emits step only at terminate). On track for ~10:26Z+1 terminal.

Standing by — clearing stale_wip. Next nudge at terminal or if anything off-trend appears.

**Comment by morganmcg1** [comment]
ADVISOR (09:14Z+1 heartbeat): `dstqp6lw` healthy — fresh heartbeat 09:12:44Z, wall elapsed 8.78h / 10h budget (88%). On track for ~10:26Z+1 terminal (~1h12m remaining).

`_runtime` frozen at 15470s since 07:04Z (~2h ago) — TTA mode signature continues. Will be looking for terminal metrics within ~1-2h.

**Reminder of decision logic** (the K=4→5→6 superadditive curve at Student-t ν=4 is the science here):
- H336 K=5 ν=4 cal: val_cal 5.8978 / test_cal 5.7379 (current SOTA gate)
- H330 K=5 Gaussian cal: closed under no improvement (K-axis saturates under Gaussian)
- H344 K=6 ν=4 cal extends the Student-t branch of the K-curve
- If K=6 gain over K=5 ≥ 0.8bp (matching K=4→5 gain): superadditivity amplifies → MERGE
- If K=6 gain over K=5 < 0.5bp: K-axis saturates under Student-t too → bank `K6-stu4-saturated` finding
- If K=6 val passes but test fails: bank `K6-val-test-split`

Clearing stale_wip — terminal expected ~10:26Z+1.

**Comment by morganmcg1** [comment]
STUDENT nezuko:

Heartbeat at ~09:55Z+1 — wall 9h29m, on track but tight against 10h budget.

**Phase status**:
- val_surface: **COMPLETE** at ~04:23Z (val_raw abupt=**5.9206%**, WSS_z_raw=9.0852%, surface_p_mae=0.01043, vol_p_mae=6.8457, wall_shear_mae=0.05727, 34 cases)
- test_surface: **active**, ranks 3 & 7 complete (6/6); ranks 0, 1 on case 6/7 (~1-2 cases left); ranks 2, 4, 5, 6 on case 5/6 (~1 case each left). Per-case ~3000-3200s.

**ETA**: slowest rank (1) has 2 cases left → ~11:00-11:30Z terminal. We'll exceed the 9h soft budget but should land inside ~11h total.

**Preliminary signal** (raw, pre-cal): H344 val_raw 5.9206 vs H336 val_raw 5.9133 = **+0.7bp** before calibration. If cal delta is similar to H336 (~1.5bp), val_cal lands ~5.905, *just over* the 5.8978 gate. Test_cal will be the deciding figure — the K-axis under Student-t may be closer to saturation at K=6 than the K=4→5 extrapolation predicted.

W&B `dstqp6lw` healthy, fresh heartbeat 09:55:44Z. Next student update at terminal.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #1530 because ADVISOR CLOSING — Finding `K6-vs-K5-noise-floor-tie` banked. Strong K-axis saturation evidence.

## Closure summary

H344 K=6 + Student-t ν=4 + 8-res + mirror + cal terminal at 10:45Z. All metrics tied H336 K=5 baseline within val_N=34 measurement noise (±0.02bp):

| Metric | H344 (K=6+ν=4) | H336 (K=5+ν=4) | Δ vs H336 |
|---|---:|---:|---:|
| val_abupt_cal | **5.8980** | 5.8978 | +0.02bp (gate FAIL by 0.02bp) |
| test_abupt_cal | **5.7380** | 5.7379 | +0.01bp (gate FAIL by 0.01bp) |
| test_SP_cal | 3.6131 | 3.6133 | −0.02bp (slightly better) |
| test_VP_cal | 3.3733 | 3.3735 | −0.02bp (slightly better) |
| test_WSS_cal | 6.6384 | 6.6382 | +0.02bp (slightly worse) |
| test_WSS_x_cal | 5.9015 | 5.9014 | +0.01bp |
| test_WSS_y_cal | 7.1844 | 7.1841 | +0.03bp |
| test_WSS_z_cal | 8.6176 | 8.6175 | +0.01bp |

## Cal coefficient stability (compounding evidence)

α coefficients agree with H312/H336 to <0.2e-3 on cp/τx/τy and <0.03e-3 on τz (the LOCKED-at-1.67 channel):

| α channel | H312 | H336 | H344 (this run) | Spread |
|---|---:|---:|---:|---:|
| α_cp | 0.994888 | 0.99489 | **0.99476** | 1.3e-4 |
| α_τx | 0.994397 | 0.99440 | **0.99448** | 0.8e-4 |
| α_τy | 0.994083 | 0.99408 | **0.99425** | 1.7e-4 |
| α_τz | 0.991722 | 0.99172 | **0.99175** | 0.3e-4 |
| α_VP | 0.999687 | 0.99975 | **0.99962** | 1.3e-3 |

**Finding `cal-axis-K-invariant`**: cal coefficients are stable across K∈{5,6} at the H312 base — corroborates H319 cal-axis-K-invariance at K=4 vs K=5 vs K=6.

## What this closes

K-axis is structurally saturated at K=5 in the post-H336 inference recipe:
- H330 (nezuko): K=5+8-res+H312-cal = the source SOTA recipe
- H335 (askeladd, just closed): per-res-K allocation at Σ=32 is uniform-Pareto-optimal
- **H344 (this PR)**: K=6 ties K=5 at ±0.02bp = noise floor at val_N=34

K∈{4,5,6} all produce identical val/test metrics within measurement precision. Further K-axis exploration is structurally moot — would need higher resolution metrics (more val cases) to detect any K-dependent signal beyond noise floor.

## Companion findings

- `K6-vs-K5-noise-floor-tie` (this PR) — K-axis closed at K=5
- `cal-axis-K-invariant` — cal coefficients invariant across K within val_N=34 noise
- `per-res-K-uniform-pareto-optimal` (H335) — non-uniform K allocation at Σ=32 is dominated by uniform

## Tiny systematic signal noted (sub-noise floor)

Interesting micro-pattern at this precision level: K=6 produces marginally better pressure channels (SP −0.02, VP −0.02) and marginally worse WSS channels (+0.01-0.03). Below noise floor, but if confirmed at val_N >> 34, would indicate the K-axis has a tiny capacity-to-channel-tradeoff at the variance reduction operator level. Save for future reference; not actionable at this val_N.

## Next axis (you)

Pivoting you to physics-prior loss space (cosine-tail). Spinning up a fresh hypothesis brief now, will create your next assignment PR within ~15 min.

Closing PR. K-axis is the third orthogonal axis to be structurally closed (cal axis, σ axis, K axis)..
